gateway-ha/src/test/java/io/trino/gateway/ha/handler/TestQueryIdCachingProxyHandler.java

```diff
@@ -33,20 +34,28 @@ public class TestQueryIdCachingProxyHandler
     public void testExtractQueryIdFromUrl()
             throws IOException
     {
-        String[] paths = {
-                "/ui/api/query/20200416_160256_03078_6b4yt",
-                "/ui/api/query/20200416_160256_03078_6b4yt?bogus_fictional_param",
-                "/ui/api/query?query_id=20200416_160256_03078_6b4yt",
-                "/ui/api/query.html?20200416_160256_03078_6b4yt"};
-        for (String path : paths) {
-            String queryId = QueryIdCachingProxyHandler.extractQueryIdIfPresent(path, null);
+        URI[] uris = {
+                URI.create("/v1/statement/executing/20200416_160256_03078_6b4yt/ya7e884929c67cdf86207a80e7a77ab2166fa2e7b/1368"),
+                URI.create("/v1/statement/queued/20200416_160256_03078_6b4yt/y0d7620a6941e78d3950798a1085383234258a566/1"),
+                URI.create("/ui/api/query/20200416_160256_03078_6b4yt"),
+                URI.create("/ui/api/query/20200416_160256_03078_6b4yt/killed"),
+                URI.create("/ui/api/query/20200416_160256_03078_6b4yt/preempted"),
+                URI.create("/v1/query/20200416_160256_03078_6b4yt?pretty"),
+                URI.create("/ui/troubleshooting?queryId=20200416_160256_03078_6b4yt"),
+                URI.create("/ui/query.html?20200416_160256_03078_6b4yt"),
+                URI.create("/login?redirect=%2Fui%2Fapi%2Fquery%2F20200416_160256_03078_6b4yt")};
+
+        for (URI uri : uris) {
```

Comment:
I would recommend removing for-loop in a preparatory commit: 
```java
        assertThat(extractQueryIdIfPresent("/ui/api/query/20200416_160256_03078_6b4yt", null))
                .isEqualTo("20200416_160256_03078_6b4yt");
        assertThat(extractQueryIdIfPresent("/ui/api/query/20200416_160256_03078_6b4yt?bogus_fictional_param", null))
                .isEqualTo("20200416_160256_03078_6b4yt");
...
```

Same for the following `nonMatchingUris` loop.

Reply:
@ebyhr I read this quickly and made the suggested change in this PR, were you suggesting to do it separately?